anta/tests/interfaces.py

```diff
@@ -883,3 +883,134 @@ def test(self) -> None:
                         output["speed"] = f"{custom_division(output['speed'], BPS_GBPS_CONVERSIONS)}Gbps"
                 failed_log = get_failed_logs(expected_interface_output, actual_interface_output)
                 self.result.is_failure(f"For interface {intf}:{failed_log}\n")
+
+
+class VerifyLACPInterfacesStatus(AntaTest):
+    """Verifies the Link Aggregation Control Protocol(LACP) status for interfaces.
+
+    - Verify that the interface is correctly added to LAG (Bundled in PortChannel).
+    - Ensure that the synchronization is established.
+    - Ensure that the interfaces are in the correct state for collecting and distributing the traffic.
+    - Validate that LACP settings, such as timeouts are correctly configured.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the provided interfaces are bundled in port channel and all specified parameters are correct.
+    * Failure: The test will fail if any interface is not bundled in port channel and any of specified parameter is not correct.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.interfaces:
+      - VerifyLACPInterfacesStatus:
+          interfaces:
+            - name: Ethernet1
+              portchannel: Port-Channel100
+    ```
+    """
+
+    name = "VerifyLACPInterfacesStatus"
+    description = "Verifies the Link Aggregation Control Protocol(LACP) status of the provided interfaces."
+    categories: ClassVar[list[str]] = ["interfaces"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaTemplate(template="show lacp interface {interface}", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifyLACPInterfacesStatus test."""
+
+        interfaces: list[Interfaces]
+        """List of interfaces with their expected state."""
+
+        class Interfaces(BaseModel):
+            """Model for an interface state."""
+
+            name: EthernetInterface
+            """Interface to validate."""
+            portchannel: PortChannel
+            """Specify PortChannel in which the interface is bundled."""
+
+    def render(self, template: AntaTemplate) -> list[AntaCommand]:
+        """Render the template for each interface in the input list."""
+        return [template.render(interface=interface.name) for interface in self.inputs.interfaces]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """Main test function for VerifyLACPInterfacesStatus."""
+        self.result.is_success()
+
+        # Iterating over command output for different interfaces
+        for command, input_entry in zip(self.instance_commands, self.inputs.interfaces):
+            interface = command.params.interface
+            portchannel = input_entry.portchannel
+
+            # Verify if a PortChannel is configured with the provided interface
+            if not (interface_details := get_value(command.json_output, f"portChannels.{portchannel}.interfaces.{interface}")):
```

Comment:
am not sure if interface can be subinterface then in that case it will break the code, please check then update the separator

Reply:
SubInterface can't be part of LACP. No need to worry about seperator. Thanks!!

Reply:
Reference: https://www.arista.com/en/um-eos/eos-ethernet-ports#subinterfaces